### PR TITLE
feat: IE8 compatibility (with a shim)

### PR DESCRIPTION
**Pull request by vvo**
This pull request adds IE8 support (still requires a polyfill on the user side).

2015-09-09-193307_1241x1076_scrot

Some of your code was already handling some IE8 things (like classList)

I know you are maybe against this PR (because you removed IE8 support) but it really does not add a
lot of code and I believe some good people will want IE8
compatibility until it is really deprecated (and we do at https://www.algolia.com/).

I prefer to try to get this merged rather than do our own fork (at
https://github.com/algolia).

Please let me know what you think!

Most of the code was about:
- prefer querySelector instead of getElementsByClassName
- handle mouse event positions for IE8
- warn the users on how to get the full IE8 support
- making the tests pass on IE8 (pass!)

See the [readme](https://github.com/leongersen/noUiSlider/blob/ad821defa33970f89b57d9a1553ba85c58793ae2/README.md#browser-support) for notes on how to support IE8.

Also added:
- easy `npm run build` command
- easy `npm run dev` command

## npm run dev

I struggled a little bit in contributing to the library (good code, no hints on how to run tests). That's no big deal and I am eager to contribute!

So I added some stack to easily provide people an `npm run dev` task
that will:

- watch src/* and rebuild
- serve tests/slider.html and automatically reload if: a test changes,
a build file change

I am sure you will love it, give it a try!

I also added:
- blanket as a dependency
- simulant to simulate click events (less code, IE8 compatible)

Thanks for your work and time :)

### CODE REVIEW

File: .npmignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+documentation/
```
Comment (vvo) [inline .npmignore]: This allows to get a lighter package in npm

File: README.md [hunk 2]
```diff
@@ -1,10 +1,15 @@
 # noUiSlider
 
-noUiSlider is lightweight JavaScript range slider, originally developed to be a jQuery UI alternative. It features cross-browser support, a wide range of options and support for a bunch of touch devices. It has been tested on Android phones, iPhone & iPad, Windows phone and touch-screen laptops and tablets and desktops; All modern browsers and IE9+ are supported. The end result? A lean, extendible and bloat-less plugin that'll just do its job.
+noUiSlider is lightweight JavaScript range slider, originally developed to be a jQuery UI alternative.
```
Comment (vvo) [inline README.md]: Took the opportunity to break some lines to make it more readable this cool paragraph

**Comment by vvo** [comment]
cc @leongersen hope you like it :)

**Comment by leongersen** [comment]
Hi Vincent! Thanks for the work on this. I didn't drop IE8 support out of spite for the browser, but simply because of the added hassle of testing for it. I'm currently traveling with no other means of connecting to the internet than my cellphone, so you wont see me merge this until early October, sorry about that. Again, thanks a bunch for contributing!

-----Original Message-----

**Comment by vvo** [comment]
No prob I knew you were remote for now, have good holidays!

**Pull request merged**